web/meetup-facilitator/src/components/PopUp.jsx

```diff
@@ -0,0 +1,38 @@
+import { useContext, useRef } from "react";
+import Button from "./Button";
+import { activitiesContext } from "../pages/Activities";
+
+const pendingPrefs = {};
+
+function PopUp({subcategories}) {
+
+    const context = useContext(activitiesContext);
+
+    const hidePopUp = () => {
+        const inputs = document.getElementsByName("activity_pref");
```

Comment:
I think it would be cleaner to use a state to keep track of each of the input values and when they change they can update the state themselves.

Reply:
I'm not exactly sure I understand how this would change things. As is, it stores the user options via userRef passed down from Activities.jsx. This function is used to persist user preferences when they exit the popup menu. And the ref is read when the popup is re-rendered to re-establish the defaultChecked status. I feel like using state to do so does the same, but adds overhead of causing a complete re-render everytime a box is checked.

Reply:
There are security reasons as well. You always want your models/data to define your views/UI, not the other way around. I can explain further on Discord or in person, but for now let's keep it as is since we're in a time crunch.